File: BOJ/1449_1.py
'''
    수리공 항승
    https://www.acmicpc.net/problem/1449
'''
import sys
input = sys.stdin.readline
sys.setrecursionlimit(1000000)
MAX = 1000 + 1
N, L = map(int, input().split())
tape = [0] * MAX
pipe = list(map(int, input().split()))
pipe.sort()


def solve():
    cnt = 0
    for pos in pipe:
        if tape[pos] == False:
            # 누수된 곳에 길이가 L인 테이프를 붙인다.
            for i in range(pos, pos + L):  # 인덱스 초과주의
                if i < MAX:
                    tape[i] = True
                else:
                    break
    # 테이프를 모두 붙인 후 테이프 갯수를 센다.
    i = 1
    while i < MAX:
        if tape[i]:
            cnt += 1
            i += (L-1)
        i += 1
    return cnt


print(solve())

File: BOJ/1449_1.cpp
// 수리공 항승
// https://www.acmicpc.net/problem/1449
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
const int MAX = 1000 + 1;
int N, L;
bool tape[MAX];
vector<int> location;
int solve()
{
    // 길이가 L인 테이프 붙이기
    for (int i = 0; i < N; i++)
    {
        int pos = location[i];
        // 물이 새는 곳에 테이프를 붙인다.
        if (!tape[pos])
        {
            for (; pos < location[i] + L; pos++)
            {
                if (pos < MAX)        // 인덱스 주의
                    tape[pos] = true; // 테이프 위치 체크
                else
                    break;
            }
        }
    }

    int cnt = 0;
    for (int i = 1; i <= MAX; i++)
    {
        // 1~1000까지 위치를 탐색하면서 테이프 갯수를 체크한다.
        if (tape[i])
        {
            cnt++;
            i += (L - 1); // L-1 만큼 이동
        }
    }
    return cnt;
}
void init()
{
    cin >> N >> L;
    location.resize(N);
    for (int i = 0; i < N; i++)
        cin >> location[i];
    sort(location.begin(), location.end());
}
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    init();
    cout << solve();
    return 0;
}